Open Judge System/Workers/OJS.Workers.ExecutionStrategies/SqlStrategies/BaseSqlExecutionStrategy.cs

```diff
@@ -21,6 +21,23 @@ public abstract class BaseSqlExecutionStrategy
 
         private const int DefaultTimeLimit = 2 * 60 * 1000;
 
+        public string WorkingDirectory { get; set; }
+
+        public ExecutionResult SafeExecute(ExecutionContext executionContext)
+        {
+            this.WorkingDirectory = DirectoryHelpers.CreateTempDirectory();
+            try
+            {
+                return this.Execute(executionContext);
+            }
+            finally
+            {
+                DirectoryHelpers.SafeDeleteDirectory(this.WorkingDirectory, true);
+            }
+        }
+
+        public abstract ExecutionResult Execute(ExecutionContext executionContext);
```

Comment:
The execution method in the inheriting strategies differs from child to child, so it cannot be made virtual in the base class.